test/lit/metadce/functions_with_names_need_escape.wat

```diff
@@ -0,0 +1,8 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py --all-items and should not be edited.
+;; RUN: wasm-metadce %s --graph-file %s.json -S -o - 
+
+(module
+  (func $f (export "f")
+     (nop)
+  )
+)
```

Comment:
Reading the full test now, I see that this does check we don't error during parse, but it doesn't check that we use the string. We do have more specific testing for JSON,

https://github.com/WebAssembly/binaryen/blob/main/test/gtest/json.cpp

Perhaps we can add a test there? Such a test could check that the output of parsing is correct too. Let me know if you want help writing such a test.